packages/projection/src/operators/Mappers/withHandles.ts

```diff
@@ -25,21 +27,34 @@ export interface WithHandles {
 const handleFromAssetId = (assetId: Cardano.AssetId) =>
   Buffer.from(Cardano.AssetId.getAssetName(assetId), 'hex').toString('utf8');
 
+const logger: Logger = dummyLogger;
+
+const isValidHandle = (handle: string) => {
+  const pattern = /^[\w.-]+$/;
+  return pattern.test(handle);
+};
+
 const getOutputHandles = (outputs: Cardano.TxOut[], policyIds: FilterByPolicyIds['policyIds']) => {
   const handles: Record<string, Handle> = {};
   for (const { address, value, datum } of outputs) {
     if (!value.assets) continue;
     for (const [assetId] of value.assets.entries()) {
       const policyId = Cardano.AssetId.getPolicyId(assetId);
       if (!policyIds.includes(policyId)) continue;
-      const handle = handleFromAssetId(assetId);
-      handles[handle] = {
-        assetId,
-        datum,
-        handle,
-        latestOwnerAddress: address,
-        policyId
-      };
+      try {
+        const handle = handleFromAssetId(assetId);
+        if (!isValidHandle(handle)) throw new HandleParsingError(handle);
+
+        handles[handle] = {
+          assetId,
+          datum,
+          handle,
+          latestOwnerAddress: address,
+          policyId
+        };
+      } catch {
+        logger.warn('Error parsing handle from assetId', { assetId });
```

Comment:
```suggestion
        logger.error(error.message);
```
Now that I see the implementation I think we can just log it as an error, since it's unexpected for the protocol.

Reply:
Fixed on e9954fc